packages/paypal-commerce-integration/src/paypal-commerce-credit-card/paypal-commerce-credit-cards-strategy.spec.ts

```diff
@@ -446,17 +288,9 @@ describe('PayPalCommerceCreditCardsPaymentStrategy', () => {
         it('submits payment without saving vaulting instrument', async () => {
             const hostedFormOrderId = 'hostedFormOrderId';
 
-            const hostedFormSubmitFnMock = jest.fn(() => ({
-                liabilityShift: undefined,
-                orderId: hostedFormOrderId,
-            }));
-
-            jest.spyOn(paypalSdk.HostedFields, 'render').mockImplementation(() =>
-                Promise.resolve({
-                    getState: () => getHostedFieldSateMock(),
-                    submit: hostedFormSubmitFnMock,
-                }),
-            );
+            // We hardcoded this private field because the only way change its value is a callback
```

Comment:
I don't think this is the only way. You can use `eventEmitter` to trigger `onApprove` wherever you want. So we'll make sure that `returnedOrderId`is set by calling this method

Reply:
You are right! Updated the PR with `eventEmitter` usage.